src/Reflection/ParametersAcceptorSelector.php

```diff
@@ -203,6 +204,32 @@ public static function selectFromArgs(
 				];
 			}
 
+			if (count($args) <= 2 && (bool) $args[0]->getAttribute(ImplodeArgVisitor::ATTRIBUTE_NAME)) {
+				$acceptor = $namedArgumentsVariants[0] ?? $parametersAcceptors[0];
+				$parameters = $acceptor->getParameters();
+				if (isset($args[1]) || ($args[0]->name !== null && $args[0]->name->name === 'array')) {
+					$parameters = [
+						new NativeParameterReflection($parameters[0]->getName(), false, new StringType(), PassedByReference::createNo(), false, null),
+						new NativeParameterReflection($parameters[1]->getName(), false, new ArrayType(new MixedType(), new MixedType()), PassedByReference::createNo(), false, null),
+					];
+				} else {
+					$parameters = [
+						new NativeParameterReflection($parameters[0]->getName(), false, new ArrayType(new MixedType(), new MixedType()), PassedByReference::createNo(), false, null),
+					];
+				}
```

Comment:
should the signature depend on php version?
https://www.php.net/manual/en/function.implode.php

<img width="559" height="432" alt="grafik" src="https://github.com/user-attachments/assets/a4bcb5a2-d6d1-4010-83d9-5fa3cb255c24" />

Reply:
Phpstan 2.1 only support php 7.4+ and currently we already have an error for the old signature (because it's deprecated)
https://phpstan.org/r/1c458f6f-d34e-433b-9f54-4d9d7e508a7d

Cf https://github.com/phpstan/phpstan-src/commit/bea339a399b7bfa2c6037eb894bc1bb88683d4b1

So I don't think I have anything more to do, unless we already change the behavior for implode on 7.4